features/lib/step_definitions/cucumber_steps.rb

```diff
@@ -32,6 +32,70 @@
   create_step_definition { string }
 end
 
+Given /^a scenario "([^\"]*)" that fails once, then passes$/ do |name|
```

Comment:
Do you think maybe these should go into a `retry_steps.rb`? They're quite specific...